src/ansiblelint/rules/unnamed_task.py

```diff
@@ -45,4 +46,5 @@ class TaskHasNameRule(AnsibleLintRule):
     def matchtask(
         self, task: Dict[str, Any], file: "Optional[Lintable]" = None
     ) -> Union[bool, str]:
-        return not task.get("name")
+        # Ignore nonamed block/always/rescue
+        return not (task.get("name") or include_nested_task_key(task))
```

Comment:
Because this rule returns a match error with the unnamed block/rescue/always if it is as it was.
If I understand correctly, we don't warn such a case currently. For e.g. in [this example](https://github.com/ansible/ansible-lint/blob/main/examples/playbooks/blockincludes.yml#L6) the blocks have no name but it is not judged as violations now. Please find [this test code](https://github.com/ansible/ansible-lint/blob/535a5b5ed3eebf9099f72ef81c629d21f29d02cb/test/test_task_includes.py#L12) as well.
```
(main)$ ansible-lint examples/playbooks/blockincludes.yml 
WARNING  Loading custom .yamllint config file, this extends our internal yamllint config.
WARNING  Listing 2 violation(s) that are fatal
unnamed-task: All tasks should be named.
examples/playbooks/tasks/directory with spaces/main.yml:3 Task/Handler: assert fail_msg=tasks in directory with spaces included

unnamed-task: All tasks should be named.
examples/playbooks/tasks/simple_task.yml:3 Task/Handler: assert fail_msg=foo

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - unnamed-task  # All tasks should be named.

Finished with 2 failure(s), 0 warning(s) on 4 files.
```

Reply:
BTW, it seems that the blocks have names and always/rescue don't in the [Ansible documentation](https://docs.ansible.com/ansible/latest/user_guide/playbooks_blocks.html).

Reply:
The name is for the combined block/rescue/always, whether or not all of them are defined. I don't know if you can have a `rescue` or an `always` without a `block`, but they all work together, not as separate tasks.

Reply:
@ssbarnea should we leave a TODO or an issue trip expand this role later? Or expand that in this PR or a follow-up?

I think it's a bug for the rule to not flag blocks without names, but I'm not sure how we have to deal with backwards compatibility here.

Reply:
If it is a bug, we also have to update some test code for example:
- test_no_same_owner_rule()
- test_included_tasks()
- test_runner_exclude_globs()
They don't expect match errors with unnamed blocks so `AssertionError` will come.

Reply:
OK. I put the bug about `unnamed-tasks` in #2104.

#2105 also lists several rules with issues that would be resolved by just using the standard `matchtask()` thanks to this PR.